File: app/src/main/java/com/spacemonster/webdataviewer/content/provider/InjectionDataProvider.java
package com.spacemonster.webdataviewer.content.provider;

import android.content.Context;

import com.spacemonster.webdataviewer.content.data.WebImage;
import com.spacemonster.webdataviewer.content.parser.JsoupWebImageParser;
import com.spacemonster.webdataviewer.content.parser.SimpleTagWebImageParser;
import com.spacemonster.webdataviewer.content.requester.FileDataRequester;
import com.spacemonster.webdataviewer.content.requester.WebDataRequester;

public class InjectionDataProvider {

    public static IDataProvider<WebImage> fileToSimpleTag(Context context, String tag){
        DataProviderImpl<String, WebImage> webImageDataProvider = new DataProviderImpl<>(context);
        webImageDataProvider.setRequester(new FileDataRequester(context));
        webImageDataProvider.setParser(new SimpleTagWebImageParser(tag));

        return webImageDataProvider;
    }

    public static IDataProvider<WebImage> webToSimpleTag(Context context, String tag){
        DataProviderImpl<String, WebImage> webImageDataProvider = new DataProviderImpl<>(context);
        webImageDataProvider.setRequester(new WebDataRequester(context));
        webImageDataProvider.setParser(new SimpleTagWebImageParser(tag));

        return webImageDataProvider;
    }

    public static IDataProvider<WebImage> webToJsoup(Context context){
        DataProviderImpl<String, WebImage> webImageDataProvider = new DataProviderImpl<>(context);
        webImageDataProvider.setRequester(new WebDataRequester(context));
        webImageDataProvider.setParser(new JsoupWebImageParser());

        return webImageDataProvider;
    }
}

File: app/src/main/kotlin/com/spacemonster/webdataviewer/content/provider/InjectionDataProvider.kt
package com.spacemonster.webdataviewer.content.provider

import android.content.Context

import com.spacemonster.webdataviewer.content.data.WebImage
import com.spacemonster.webdataviewer.content.parser.JsoupWebImageParser
import com.spacemonster.webdataviewer.content.parser.SimpleTagWebImageParser
import com.spacemonster.webdataviewer.content.requester.FileDataRequester
import com.spacemonster.webdataviewer.content.requester.WebDataRequester

object InjectionDataProvider {

    fun fileToSimpleTag(context: Context, tag: String, dataPath: String): IDataProvider<String, WebImage> {
        return DataProviderImpl(FileDataRequester(context), SimpleTagWebImageParser(tag), dataPath)
    }

    fun webToSimpleTag(context: Context, tag: String, dataPath: String): IDataProvider<String, WebImage> {
        return DataProviderImpl(WebDataRequester(context), SimpleTagWebImageParser(tag), dataPath)
    }

    fun webToJsoup(context: Context, dataPath: String): IDataProvider<String, WebImage> {
        return DataProviderImpl(WebDataRequester(context), JsoupWebImageParser(), dataPath)
    }
}
